src/axom/mint/docs/sphinx/sections/tutorial.rst

```diff
@@ -0,0 +1,1255 @@
+
+   * The field name, by calling ``getName``,
+   * The number of tuples of the field, by calling ``getNumTuples()``
+   * The number of components of the field, by calling ``getNumComponents()``
+   * The underlying field type by calling ``getType()``
+
+* Given the above metadata, the application can then obtain a pointer to the
+  raw field data by calling ``getFieldPtr()`` on the target mesh, as shown in
+  the code snippet above.
+
+.. _usingExternalStorage:
+
+Using External Storage
+^^^^^^^^^^^^^^^^^^^^^^^
+
+A Mint mesh may also be constructed from :ref:`ExternalStorage`. In this
+case, the application holds buffers that describe the constituent
+:ref:`Geometry`, :ref:`Topology` and :ref:`FieldData` of the mesh, which, are
+wrapped in Mint for further processing.
+
+The following code snippet illustrates how to use :ref:`ExternalStorage` using
+the :ref:`SingleCellTopology` :ref:`UnstructuredMesh` used to demonstrate
+how to :ref:`createAnUnstructuredMesh` with :ref:`NativeStorage`:
+
+.. literalinclude:: ../../../examples/mint_tutorial.cpp
+   :start-after: sphinx_tutorial_using_external_storage_start
+   :end-before: sphinx_tutorial_using_external_storage_end
+   :language: C++
+   :linenos:
+
+* The application has the following buffers:
+
+  * ``x`` and ``y`` buffers to hold the coordinates of the :ref:`Nodes`
+  * ``cell_connectivity``, which, stores the *cell-to-node* connectivity
+  * ``den`` which holds a *scalar density* field defined over the constituent
+    :ref:`Cells` of the mesh.
+
+* The mesh is created by calling the ``mint::UnstructuredMesh`` class
+  constructor, with the following arguments:
+
+  * The cell type, i.e, ``mint::TRIANGLE``,
+  * The total number of cells, ``NUM_CELLS``,
+  * The ``cell_connectivity`` which specifies the :ref:`Topology` of the
+    :ref:`UnstructuredMesh`,
+  * The total number of nodes, ``NUM_NODES``, and
+  * The  ``x``, ``y`` coordinate buffers that specify the :ref:`Geometry` of
+    the :ref:`UnstructuredMesh`.
+
+* The *scalar density* field is registered with Mint by calling the
+  ``createField()`` method on the target mesh instance, as before, but also
+  passing the raw pointer to the application buffer in a third argument.
+
+.. note::
+
+   The other :ref:`MeshTypes` can be similarly constructed using
+   :ref:`ExternalStorage` by calling the appropriate constructor.
+   Consult the `Mint Doxygen API Documentation`_ for more details.
+
+The resulting mesh instance points to the application's buffers. Mint may be
+used to process the data e.g., :ref:`outputToVTK`
+etc. The values of the data may also be modified, however, the mesh cannot
+dynamically grow or shrink when using :ref:`ExternalStorage`.
+
+.. warning::
+
+   A mesh using :ref:`ExternalStorage` may modify the values of the application
+   data. However, the data is owned by the application that supplied the
+   external buffers. Consequently, the associated memory cannot be reallocated
```

Comment:
Please clarify to say "Mint cannot reallocate external buffers to grow or shrink the mesh."